### ISSUE TITLE
Method Category::getCategoriesWithoutParent is wrong

### ISSUE THREAD

**Issue by StoreCommanderDev** [PrestaShop/PrestaShop#37374]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

The methode name is : `getCategoriesWithoutParent`
The comment is : `Get Categories without parent.`

The WHERE condition is:

```sql
WHERE `level_depth` = 1');
``` 

There is no condition on `id_parent` field.

In database there are one or more categories with level_depth =1 and id_parent =1 for example.

When I call `$category->getParentsCategories()` (this method use the `Category::getCategoriesWithoutParent()` in it) instead of stopping at **Home category** it will go back to **Root category**

### Expected behavior

When I call `$category->getParentsCategories()` it doesn't must provide the Root category

### Steps to reproduce

Add (with code) a category with level_depth = 1

### NOK

1. calling `(new Category(X))->getParentsCategories()`
2. looking at result and **find Root category** in it

![Image](https://github.com/user-attachments/assets/32b7d230-1e78-4197-b4ba-3907f443b5bc)

### OK
1. adding id_parent = 0 to getCategoriesWithoutParent() method
2. calling `(new Category(X))->getParentsCategories()`
3. looking at result and **no Root category** in it
![Image](https://github.com/user-attachments/assets/673a39d2-f487-43e2-81b1-c3b6378f0186)

### PrestaShop version(s) where the bug happened

8.2

### PHP version(s) where the bug happened

8.1

**Comment by Hlavtox** [comment]
What is the problem here? I don't understand.

`getCategoriesWithoutParent` select all level 1 categories from the shop = all the HOME categories. Without the ROOT category with ID 1.
`(new Category(x))->getParentsCategories()` selects all parent categories of the current category. Without the ROOT category with ID 1.

Why should it contain the root category? The category tree on the current shop starts with the HOME category, with ID 2, usually.

**Comment by StoreCommanderDev** [comment]
In the case of you have created one or more categories with a level 1 and id_parent =1, `getParentsCategories()` contains root Category

In this case `getCategoriesWithoutParent()` return more than 1 row.

in `getParentsCategories()`, there is
```php
if (count(Category::getCategoriesWithoutParent()) > 1) {
    $context->shop->id_category = (int) Configuration::get('PS_ROOT_CATEGORY');
}
``` 

So in the last sql request the `$context->shop->id_category` became ROOT 
```php
$rootTreeInfo = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow(
    'SELECT c.`nleft`, c.`nright` FROM `' . _DB_PREFIX_ . 'category` c
WHERE c.`id_category` = ' . (int) $context->shop->id_category
);

$categories = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
    'SELECT c.*, cl.*  ' . $sqlAppend .
    ' WHERE c.`nleft` <= ' . (int) $treeInfo['nleft'] .
    ' AND c.`nright` >= ' . (int) $treeInfo['nright'] .
    ' AND c.`nleft` >= ' . (int) $rootTreeInfo['nleft'] .
    ' AND c.`nright` <= ' . (int) $rootTreeInfo['nright'] .
    ' ORDER BY `nleft` DESC'
);
``` 

So `$categories` will contains all categories between **ROOT nleft & nright** included itself

**Comment by StoreCommanderDev** [comment]
Hello everyone.

For more information, I reproduced by using back office.

In multi-store context, PrestaShop allow to add new Root Category in case you need to have a specific category for a shop.

If I create a new root category for my shop number 3 (or else), a new category is created with a level_depth = 1.

So the real root appear by using the method `getParentsCategories()
`
![Image](https://github.com/user-attachments/assets/61642345-cd28-4105-96ca-52cf0895897d)

**Comment by Hlavtox** [comment]
I think you are confusing ROOT and HOME category.

Unless I speak nonsense (@kpodemski please confirm), the main category in multistore context has level_depth = 1.

**Comment by kpodemski** [comment]
I agree with @StoreCommanderDev . The method is a bit confusing, I have this structure:

Root
Root -> B2B Home (ID 2) -> Some Products Category
Root -> B2C Home (ID 10) -> Some Products Category (ID 11)

When I run this code:

```
$category = new Category(10);
$category->getParentsCategories();
```

I would expect either no results or only the root category. It gives me the root category and the B2C Home category. 


But, as long as I understand the confusion, I'm afraid any changes to this method may produce unexpected results for third-party solutions. The last time `Category::getCategoriesWithoutParent` was modified over 11 years ago.

**Comment by Hlavtox** [comment]
getParentsCategories() just returns all parents including itself, it's not that hard to understand it.

I use it for example here - https://github.com/PrestaShop/PrestaShop/blob/576ed136289fef063a5f3777b6731b96ec8d434d/controllers/front/listing/CategoryController.php#L362

If you want just the parent - `$category['id_category'] != $this->category->id`

**Comment by StoreCommanderDev** [comment]
Some module editor build some tree with getParentsCategories() method.

So one day it start with Home > categ 1 > categ 2

And another day, after modifying categories structure (for business)

Method return Root > Home > categ1 > categ 2 

So it can impact their own module

**Comment by web-cooking-factory** [comment]
Hi, from my point of view, it's more a context issue in a way. We can see in getParentsCategories that later there is a check on id_parent if in shop context but we can't set that shop context. So add a parameter with id_shop and use it to set context if not null seems more safe to me.

The condition count(Category::getCategoriesWithoutParent()) > 1) is just there to ensure there is a root to the tree. In shop context, it has to be the home category of the shop, otherwise it has to be the root category.

But in the case of shop group context, the result should not include the tree of others shop groups.

So there's definitely something to do.

**Comment by paulnoelcholot** [comment]
Hello @StoreCommanderDev ,

Thanks for reporting this issue!
Ping @PrestaShop/prestashop-core-developers : Could someone please try to reproduce the issue, it's too technical!

Thanks in advance!

**Comment by jolelievre** [comment]
Hi everyone,

so just to give some context and summarize the conversation

First the condition `level = 1` is completely valid, the Category tree data is based on the [Nest set model](https://en.wikipedia.org/wiki/Nested_set_model) that computes right/left boundaries and the depth This is an algorithm that allows handling huge levels in a tree efficiently
In a typical nested tree to get all the root nodes (without parent) we would fetch all the nodes which depth is 0, in the PrestaShop category structure we have this `Root` category that contains all of them, I don't know why this is needed exactly but if we consider this category needs to be filtered out and ignored then the root nodes' depth becomes 1, which is why the condition makes sense therefore the content of `getCategoriesWithoutParent` doesn't seem to need modification IMHO.

So we should focus on `getParentsCategories`.

Now about this special `Root` category, I think the problem is not if there is a condition `id_parent` or not, because the condition added in `$sqlAppend` is not even related to how far up we go in the tree anyway. The main point is in this query:

```php
$rootTreeInfo = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow(
                'SELECT c.`nleft`, c.`nright` FROM `' . _DB_PREFIX_ . 'category` c
            WHERE c.`id_category` = ' . (int) $context->shop->id_category
            );
```

that fetches the root category, or at least what is considered as the root category in this context, and that's what defines the limit of the returned breadcrumb.

In usual cases it uses the Shop default category so the breadcrumb can't return more than shop's content which seems fine.

But when we are ni All shops or Shop group context then it makes sense that we need to fetch more categories, the main reason being that if you want to access all the "Root categories" (and I mean by that the ones with depth=1 that are just under the Root(id=1) category) then you need to be able to go back up to this "Root" category to then go down into all the other ones in the tree.

But it probably is only useful in the back office, and only when you are not in single shop context, so I have a suggestion let me know if you think it can create unexpected side effects:

Transofmr this:

```php
if (count(Category::getCategoriesWithoutParent()) > 1) {
            $context->shop->id_category = (int) Configuration::get('PS_ROOT_CATEGORY');
        }
```

into this:

```php
        if (Shop::getContext() !== Shop::CONTEXT_SHOP && count(Category::getCategoriesWithoutParent()) > 1) {
            $context->shop->id_category = (int) Configuration::get('PS_ROOT_CATEGORY');
        }
```

So the "Root" category would only be displayed when you are in All shops or in Shop group context. I'm not sure we need to handle shop group differently because it would turn overly complex to find which category is a root for all the shops inside the group.

What do you guys think? I'm gonna just create a small draft PR to run the tests and see if it has direct impact, although this won't be a guaranty it doesn't create bugs because I'm not sure we have a strong coverage in multi shop, but that's a first check.

**Comment by jolelievre** [comment]
Here is the PR to test the solution https://github.com/PrestaShop/PrestaShop/pull/38168

**Comment by jolelievre** [comment]
@kpodemski @StoreCommanderDev @Hlavtox what do you think?

**Comment by StoreCommanderDev** [comment]
Thank you for your message @jolelievre  and the time spent on the subject.
In my opinion it seems good. It will need some tests to be sure its all ok without regression :)
